CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+
+    Please try to make sure that your enhancement is distinct from any others that have already been requested or implemented.
+    If you find one that's similar but there are subtle differences please reference the other request in your issue.
+
+## Making a change
+
+We appreciate all contributions to `jupyter-book`, but those accepted fastest will follow a workflow similar to the following:
+
+**1. Comment on an existing issue or open a new issue referencing your addition.**
+
+This allows other members of the jupyter-book development team to confirm that you aren't overlapping with work that's currently underway and that everyone is on the same page with the goal of the work you're going to carry out.
+
+[This blog][link_pushpullblog] is a nice explanation of why putting this work in up front is so useful to everyone involved.
+
+**2. [Fork][link_fork] the [jupyter-book repository][link_jupyter-book] to your profile.**
+
+This is now your own unique copy of jupyter-book.
+Changes here won't effect anyone else's work, so it's a safe space to explore edits to the code!
+
+Make sure to [keep your fork up to date][link_updateupstreamwiki] with the master repository.
+
+**3. Make the changes you've discussed.**
+
+Try to keep the changes focused.
+We've found that working on a [new branch][link_branches] makes it easier to keep your changes targeted.
+
+**4. Submit a [pull request][link_pullrequest].**
+
+A member of the development team will review your changes to confirm that they can be merged into the main code base.
+When opening the pull request, we ask that you follow some [specific conventions](#pull-requests).
+We outline these below.
+
+### Pull Requests
+
+To improve understanding pull requests "at a glance", we encourage the use of several standardized tags.
+When opening a pull request, please use at least one of the following prefixes:
+
+* **[BRK]** for changes which break existing builds or tests
+* **[DOC]** for new or updated documentation
+* **[ENH]** for enhancements
+* **[FIX]** for bug fixes
+* **[REF]** for refactoring existing code
+* **[STY]** for stylistic changes
+* **[TST]** for new or updated tests, and
+* **[WIP]** for changes which are not yet ready to be merged
+
+Pull requests should be submitted early and often!
+If your pull request is not yet ready to be merged, please also include the **[WIP]** prefix.
+This tells the development team that your pull request is a "work-in-progress",
+and that you plan to continue working on it.[link_kanban]: https://en.wikipedia.org/wiki/Kanban_board
+
+
+You can also combine the tags above, for example if you are updating both a test and
+the documentation: **[TST, DOC]**.
+If you're still working on the pull request that prefix would be **[WIP, TST, DOC]**.
+
+
+## Recognizing contributors
+
+We welcome and recognize all contributions from documentation to testing to code development.
+You can see a list of current contributors in the [contributors tab][link_contributors].
```

Comment:
What's the process by which somebody becomes a contributor? I think I'm fine with just growing the list very liberally (e.g. after someone has been a regular commentor, or after one or two PRs). What do you think about this?

Reply:
Right now I have it just pointing to the [contributor github tab](https://github.com/jupyter/jupyter-book/graphs/contributors). The downside to this is that it doesn't actually recognize any contributions that aren't code or docs !

If we create a separate `humans.txt` (or zenodo or something similar), I'd say that it's hard to come up with firm guidelines across all the different possible contribution types, so maybe encourage folks to add their name after some sustained or significant engagement ? TBD what that means to you.

Reply:
fair enough - how about we cross the bridge of "offically listing contributors" later on...I think that's a complex topic and I don't want it to block this big improvement in suggesting to folks how they can get involved!